Orange/data/io.py

```diff
@@ -618,15 +618,25 @@ def data_table(cls, data, headers=None):
         # Determine maximum row length
         rowlen = max(map(len, (names, types, flags)))
 
+        strip = False
+
         def _equal_length(lst):
-            lst.extend(['']*(rowlen - len(lst)))
+            nonlocal strip
+            if len(lst) > rowlen > 0:
+                lst = lst[:rowlen]
+                strip = True
+            elif len(lst) < rowlen:
+                lst.extend(['']*(rowlen - len(lst)))
             return lst
 
         # Ensure all data is of equal width in a column-contiguous array
         data = [_equal_length([s.strip() for s in row])
                 for row in data if any(row)]
         data = np.array(data, dtype=object, order='F')
 
+        if strip:
+            warnings.warn("Columns with no headers were striped.")
```

Comment:
I would prefer "Columns with no headers were removed.", since stripped is a bit Pythonish. Or at least 'stripped'.